lib/srv/db/mysql/engine.go

```diff
@@ -330,3 +356,55 @@ func (e *Engine) makeAcquireSemaphoreConfig(sessionCtx *common.Session) services
 		},
 	}
 }
+
+// appendGCPClientCert calls the GCP API to generate an ephemeral certificate
+// when the instance's RequireSsl setting is true. Return requireSSL so the
+// caller can configure a custom dialer.
+func (e *Engine) appendGCPClientCert(ctx context.Context, sessionCtx *common.Session, tlsConfig *tls.Config) (requireSSL bool, err error) {
+	svc, err := e.CloudClients.GetGCPSQLAdminClient(ctx)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+
+	dbi, err := svc.GetDatabaseInstance(ctx, sessionCtx)
```

Comment:
I think this might break access for existing setup which don't have the "get database instance" permission. I think it's a plausible scenario as we haven't been requiring it until now. Can we handle permission errors gracefully and fallback to "not generate client cert" (basically, keep existing behavior) in that case?

Reply:
Added fallback. Tested with limited permissions and everything worked with the fallback. I'd like to test once again though just to be sure. Getting the IAM permissions right is no fun.